python/ray/experimental/compiled_dag_ref.py

```diff
@@ -102,10 +113,39 @@ def get(self, timeout: Optional[float] = None):
             )
 
         self._ray_get_called = True
-        return_vals = self._dag._execute_until(
-            self._execution_index, self._channel_index, timeout
-        )
-        return _process_return_vals(return_vals, True)
+        try:
+            return_vals = self._dag._execute_until(
+                self._execution_index, self._channel_index, timeout
+            )
+            return _process_return_vals(return_vals, True)
```

Comment:
Can you move `_process_return_vals` out of the try block? (better to minimize region where exceptions could be caught).